service/worker/migration/activities.go

```diff
@@ -51,6 +53,65 @@ import (
 	"go.temporal.io/server/common/util"
 )
 
+type (
+	VerifyStatus int
+	VerifyResult struct {
+		Status VerifyStatus
+		Reason string
+	}
+
+	replicationTasksHeartbeatDetails struct {
+		Results                       []VerifyResult
+		CheckPoint                    time.Time
+		LastNotFoundWorkflowExecution commonpb.WorkflowExecution
```

Comment:
What is the purpose of this? Should we use a fixed length slice?

Reply:
VerifyResult keeps the status of replication tasks for each execution task from input, which is a variable-length array: